### ISSUE TITLE
crono service doesn't autorun on docker-compose

### ISSUE THREAD

**Issue by gbisheimer** [SUSE/Portus#418]
I found this great project (excellent work btw) and was testing this project using the docker-compose way.
After struggling a little with the hostname and registry config (#417 and #415) I was able to get it working on my server, but cant get local db to get synced with the configured registry.
I have read the [docs on syncing](https://github.com/SUSE/Portus/wiki/Synchronizing-the-Registry-and-Portus) and found that cronos service was not running. I have to start it manually to get the sync working. Also, couldn't get registry notifications working either to update Portus DB.

There is anything I'm need to configure to get this working automatically?

Thanks!

**Comment by wangchunyang** [comment]
Once I pushed an image to registry, the notification is indeed received by portus_web. However an error occurred when portus tried to fetch tags by digest hash. We can see the logs as follow:

```

Started POST "/v2/webhooks/events" for 15.209.122.238 at 2015-10-12 07:48:21 +0000
Cannot render console from 15.209.122.238! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
Processing by Api::V2::EventsController#create as JSON
  Registry Load (0.3ms)  SELECT  `registries`.* FROM `registries` WHERE `registries`.`hostname` = '172.17.42.1:5000' LIMIT 1
  Namespace Load (0.4ms)  SELECT  `namespaces`.* FROM `namespaces` WHERE `namespaces`.`registry_id` = 1 AND `namespaces`.`name` = 'chunyangwanghpecom' LIMIT 1
Could not fetch the tag for target {"mediaType"=>"application/vnd.docker.distribution.manifest.v1+json", "size"=>3226, "digest"=>"sha256:767b41991f058a23c75400a6b6b2476f5b3c5474ca8483091123419c36727973", "length"=>3226, "repository"=>"chunyangwanghpecom/busybox", "url"=>"http://172.17.42.1:5000/v2/chunyangwanghpecom/busybox/manifests/sha256:767b41991f058a23c75400a6b6b2476f5b3c5474ca8483091123419c36727973"}
Reason: Net::ReadTimeout
Completed 202 Accepted in 120153ms (ActiveRecord: 0.7ms)
```
From the logs above, we can see the URL of registry is http://172.17.42.1:5000. It looks this URL is not accessible from portus web container.

**Comment by wangchunyang** [comment]
Please note that the output in logs - "Reason: Net::ReadTimeout". However I can ping it from container as follow:
```
$ sudo docker exec -it portus_web_1 ping 172.17.42.1

PING 172.17.42.1 (172.17.42.1): 56 data bytes
64 bytes from 172.17.42.1: icmp_seq=0 ttl=64 time=0.125 ms
64 bytes from 172.17.42.1: icmp_seq=1 ttl=64 time=0.204 ms
64 bytes from 172.17.42.1: icmp_seq=2 ttl=64 time=0.284 ms
64 bytes from 172.17.42.1: icmp_seq=3 ttl=64 time=0.132 ms
64 bytes from 172.17.42.1: icmp_seq=4 ttl=64 time=0.198 ms
```

**Comment by flavio** [comment]
@wangchunyang this issue is about another topic. The problem you are describing is a duplicate of issue https://github.com/SUSE/Portus/issues/376, take a look at [this comment](https://github.com/SUSE/Portus/issues/376#issuecomment-146348415). We have a working nginx configuration in there that can be used to prevent this kind of issues: https://github.com/SUSE/Portus/wiki/An-NGinx-setup

**Comment by wangchunyang** [comment]
@flavio thanks for your reply. I will try NGinx option. However, please note that I'm just running ./compose-setup.sh to run three containers for testing (portus_web_1, portus_db_1, portus_registry_1). Do I still need to setup NGinx to run it? The hostname is "172.17.42.1:5000" is just from the network interface docker0.

**Comment by flavio** [comment]
@wangchunyang I'm working on an update of our compose setup to fix this kind of issues, gimme a couple of hours.

About the hostname issue, this has just been fixed a couple of minutes ago with a PR.

Please wait till I have the 1st change into master.

**Comment by wangchunyang** [comment]
@flavio thanks. I already used the PR to fix the hostname issue in compose-setup.sh. Then I got the "Reason: Net::ReadTimeout" issue during getting tag in notification event handler. As you mentioned, I will monitor another issue #423. Thanks for your quick response.

**Comment by mssola** [comment]
Fixed with the commit ca0a1f1f1a903a15aa0fba45e3efde5e889811ad

**Comment by flavio** [comment]
This is solved. Note well you will probably have to rebuild the web image to have some changes. I don't know if compose is smart enough...
